### ISSUE TITLE
'cmus' terminated by signal SIGSEGV (Address boundary error)

### ISSUE THREAD

**Issue by bennesp** [cmus/cmus#500]
Have a problem similar to #154, but at first launch.
Tried to purge the ubuntu package, and install it from source, without success.

The error I received is:

```
fish: Job 1, 'cmus' terminated by signal SIGSEGV (Address boundary error)
```

---

1) `uname -a`:
```
Linux localhost 3.10.18 #1 SMP Tue Aug 2 21:20:02 PDT 2016 x86_64 x86_64 x86_64 GNU/Linux
```

2) `cmus --version`:
```
cmus v2.7.0-144-g37f4daa
Copyright 2004-2006 Timo Hirvonen
Copyright 2008-2013 Various Authors
```

3) `cmus --plugins`:
```
Input Plugins: /usr/local/lib/cmus/ip
  wav:
    Priority: 50
    File Types: wav
    MIME Types:

Output Plugins: /usr/local/lib/cmus/op
  alsa
  oss
```

**Comment by bennesp** [comment]
~/cmus-debug.txt

```
main: charset = 'ANSI_X3.4-1968'
player_init: using real-time scheduling
player_init: setting priority to 99
player_set_op: selecting first initialized op
init_plugin: initialized mixer for alsa
init_plugin: initialized pcm for alsa
alsa_set_hw_params: can pause = 0
alsa_set_hw_params: rate=44100
find_mixer_elem_by_name: name = Master
find_mixer_elem_by_name: has playback volume = 1
find_mixer_elem_by_name: has playback switch = 1
find_mixer_elem_by_name: name = Capture
find_mixer_elem_by_name: has playback volume = 0
find_mixer_elem_by_name: has playback switch = 0
alsa_mixer_open: mixer element `PCM' not found, trying `Master'
find_mixer_elem_by_name: name = Master
find_mixer_elem_by_name: has playback volume = 1
find_mixer_elem_by_name: has playback switch = 1
```

**Comment by flyingmutant** [comment]
Can you please launch the version you've built under gdb/lldb, and post a backtrace?

**Comment by bennesp** [comment]
```
(gdb) r
Starting program: /home/bennes/Downloads/cmus/cmus 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff639c700 (LWP 24499)]
[New Thread 0x7ffff5b9b700 (LWP 24500)]
[New Thread 0x7ffff539a700 (LWP 24501)]
[New Thread 0x7ffff4584700 (LWP 24502)]

Program received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007ffff67e07de in snd_hctl_handle_events () from /usr/lib/x86_64-linux-gnu/libasound.so.2
#2  0x00007ffff67eb971 in snd_mixer_handle_events () from /usr/lib/x86_64-linux-gnu/libasound.so.2
#3  0x00007ffff6a9a262 in alsa_mixer_get_volume (l=0x650484 <volume_l>, r=0x650480 <volume_r>) at mixer_alsa.c:175
#4  0x000000000040b4b5 in init_all () at ui_curses.c:2315
#5  main (argc=<optimized out>, argv=<optimized out>) at ui_curses.c:2464
(gdb)
```

**Comment by mahkoh** [comment]
Compile alsa with debug symbols and try again.

**Comment by bennesp** [comment]
This sounds weird.
Removing alsa and installing the debug version made the trick, cmus now working.
Then I uninstalled libasound-dbg and installed again the libasound (without debug) and cmus working again. 👍 
Checked for SHA1 on libasound.so.2.0.0 but it's the same of before. Don't know how, but now it's opening.

The fact is that now ```cmus``` does't add neither mp3 or ogg files, but that's another problem.
Thanks for the help
